zenoh-plugin-ros2dds/src/route_service_cli.rs

```diff
@@ -406,51 +408,21 @@ fn route_dds_request_to_zenoh(
                 },
             }
         })
-        .res_sync()
+        .wait()
     {
         tracing::warn!("{route_id}: routing request {request_id} from DDS to Zenoh failed: {e}");
     }
 }
 
-// TODO: remove and replace with Zenoh's CallbackPair when https://github.com/eclipse-zenoh/zenoh/pull/653 is available
-struct CallbackPair<Callback, DropFn>
-where
-    DropFn: FnMut() + Send + Sync + 'static,
-{
-    pub callback: Callback,
-    pub drop: DropFn,
-}
-
-impl<Callback, DropCallback> Drop for CallbackPair<Callback, DropCallback>
-where
-    DropCallback: FnMut() + Send + Sync + 'static,
-{
-    fn drop(&mut self) {
-        (self.drop)()
-    }
-}
-
-impl<'a, OnEvent, Event, DropCallback> IntoCallbackReceiverPair<'a, Event>
-    for CallbackPair<OnEvent, DropCallback>
-where
-    OnEvent: Fn(Event) + Send + Sync + 'a,
-    DropCallback: FnMut() + Send + Sync + 'static,
-{
-    type Receiver = ();
-    fn into_cb_receiver_pair(self) -> (Callback<'a, Event>, Self::Receiver) {
-        (Dyn::from(move |x| (self.callback)(x)), ())
-    }
-}
-
 fn route_zenoh_reply_to_dds(
     route_id: &str,
     reply: Reply,
     request_id: CddsRequestHeader,
     rep_writer: dds_entity_t,
 ) {
-    match reply.sample {
+    match reply.result() {
         Ok(sample) => {
-            let zenoh_rep_buf = sample.payload.contiguous();
+            let zenoh_rep_buf = sample.payload().into::<Vec<u8>>();
```

Comment:
Use `Cow<[u8]>` whenever it is possible.

```suggestion
            let zenoh_rep_buf = sample.payload().into::<Cow<[u8]>>();
```